test/output-app/model/Tweet.js

```diff
@@ -1,60 +1,158 @@
-import DataLoader from 'dataloader';
-import findByIds from 'mongo-find-by-ids';
+/* eslint-disable prettier */
+import {
+  queryForRoles,
+  onAuthRegisterLoader,
+  authlog,
+  checkAuthDoc
+} from 'create-graphql-server-authorization';
 
 export default class Tweet {
   constructor(context) {
     this.context = context;
     this.collection = context.db.collection('tweet');
     this.pubsub = context.pubsub;
-    this.loader = new DataLoader(ids => findByIds(this.collection, ids));
+    const { me, User } = context;
+    queryForRoles(
+      me,
+      ['admin', 'world'],
+      ['authorId', 'coauthorsIds'],
+      { User },
+      onAuthRegisterLoader('tweet findOneById', 'readOne', me, this)
+    );
```

Comment:
I don't understand this part -- why does it need to be async (or callback-based)?

Is it because `onAuthRegisterLoader` ["doesn't throw exception in error case"](https://tobkle.github.io/create-graphql-server-authorization/function/index.html#static-function-onAuthRegisterLoader)? What is the error case? Why is it legitimate?